qiskit/qpy/binary_io/circuits.py

```diff
@@ -347,6 +347,11 @@ def _read_instruction(
                 else:
                     gate = gate_class(*params)
                     gate.label = label
+            # NOTE: MSGate is deprecated, so we will switch it to the GMS gate. See
+            #       https://github.com/Qiskit/qiskit/issues/11378.
+            if gate_name == "MSGate":
```

Comment:
using `if` in here change the logic of this condition.
The else in Line355 now will be same as `if label is not None and gate_name != "MSGate"`
In my opinion, you should use `elif`
My proposal code for Line352: `elif gate_name == "MSGate":`